### PR TITLE
Mail: Customize activity types in apps menus

### PR DESCRIPTION
**Pull request by mba-odoo**
Description of the issue/feature this PR addresses:
Task : https://www.odoo.com/web#id=31881&view_type=form&model=project.task&action=327&menu_id=4720
Pad : https://pad.odoo.com/p/r.4db8bad243c15dfaf7387ca460b28b67

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sales_team/views/crm_team_views.xml [hunk 1]
```diff
@@ -196,5 +196,14 @@
...
             action="sales_team_config_action"/>
 
+        <record id="sales_team_config_mail_activity_type_action" model="ir.actions.act_window">
+            <field name="name">Activity Types</field>
+            <field name="res_model">mail.activity.type</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">tree,form</field>
+            <field name="domain">['|', ('res_model_id','=', False), ('res_model_id','in',['crm.lead', 'res.partner', 'sale.order', 'product.template', 'product.product'])]</field>
+        </record>
+        <menuitem name="Activity Type" id="menu_sales_team_config_mail_activity_type" parent="sales_team.menu_sale_config" action="sales_team_config_mail_activity_type_action" sequence="2"/>
```
Comment (bst-odoo) [inline addons/sales_team/views/crm_team_views.xml]: Menu item should be "Activity Types"

Comment (mba-odoo) [inline addons/sales_team/views/crm_team_views.xml]: @bst-odoo  Done

File: addons/mail/views/mail_activity_views.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
                     </div>
                     <group>
                         <group>
-                            <field name="res_model_id"/>
+                            <field name="res_model_id" groups="base.group_no_one"/>
```
Comment (tde-banana-odoo) [inline addons/mail/views/mail_activity_views.xml]: Please move that in a separate commit. As we target a stable, each separate point should be in a separate commit.

File: addons/project/views/project_views.xml [hunk 3]
```diff
@@ -790,6 +790,15 @@ action = model.activate_sample_project()
         </record>
         <menuitem action="project_tags_action" id="menu_project_tags_act" parent="menu_project_config" groups="base.group_no_one"/>
 
+        <record id="project_config_mail_activity_type_action" model="ir.actions.act_window">
+            <field name="name">Activity Types</field>
+            <field name="res_model">mail.activity.type</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">tree,form</field>
+            <field name="domain">['|', ('res_model_id','=', False), ('res_model_id','in',['project.task', 'project.issue'])]</field>
```
Comment (tde-banana-odoo) [inline addons/project/views/project_views.xml]: You should have a domain on task in project and add issues in project_issue. We try to enforce modularity.

File: addons/sales_team/views/crm_team_views.xml [hunk 4]
```diff
@@ -196,5 +196,14 @@
             parent="sales_team.menu_sale_config"
             action="sales_team_config_action"/>
 
+        <record id="sales_team_config_mail_activity_type_action" model="ir.actions.act_window">
+            <field name="name">Activity Types</field>
+            <field name="res_model">mail.activity.type</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">tree,form</field>
+            <field name="domain">['|', ('res_model_id','=', False), ('res_model_id','in',['crm.lead', 'res.partner', 'sale.order', 'product.template', 'product.product'])]</field>
```
Comment (tde-banana-odoo) [inline addons/sales_team/views/crm_team_views.xml]: This one is a tricky one. Probably in crm you should set a domain in lead module on lead and partner; one in sales on sale order and product; and one in sale_crm that takes both of them. Do you think this could enforce modularity and still work as requested ?

File: addons/mail/views/mail_activity_views.xml (hunk 2 above)
Comment (mba-odoo) [inline addons/mail/views/mail_activity_views.xml]: @tde-banana-odoo , Done.
